test/zipkin/util.go

```diff
@@ -65,26 +69,66 @@ var (
 	teardownOnce sync.Once
 )
 
+// SetupZipkinTracingFromConfigTracing setups zipkin tracing like SetupZipkinTracing but retrieving the zipkin configuration
+// from config-tracing config map
+func SetupZipkinTracingFromConfigTracing(kubeClientset *kubernetes.Clientset, logf logging.FormatLogger, configMapNamespace string) error {
+	cm, err := kubeClientset.CoreV1().ConfigMaps(configMapNamespace).Get("config-tracing", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Error while retrieving config-tracing config map: %w", err)
+	}
+	c, err := tracingconfig.NewTracingConfigFromConfigMap(cm)
+	if err != nil {
+		return fmt.Errorf("Error while parsing config-tracing config map: %w", err)
+	}
+	zipkinEndpointUrl, err := url.Parse(c.ZipkinEndpoint)
+	if err != nil {
+		return fmt.Errorf("Error while parsing the zipkin endpoint in config-tracing config map: %w", err)
+	}
+	unparsedPort := zipkinEndpointUrl.Port()
+	port := 80
+	if unparsedPort != "" {
+		port, err = strconv.Atoi(unparsedPort)
```

Comment:
`strconv.ParseUint(unparsedPort, 10, 16` — will ensure a lot of edgecases are covered.